### PR TITLE
[MCOMPILER-472] Update plugin (requires Maven 3.2.5+)

### PR DESCRIPTION
**Pull request by cstamas**
Set proper scopes for maven deps, drop cruft not
needed since eons. We do not test against maven 3.0
or maven 2 anymore, so upped testing harness (was 2.x!)
and hence, upped maven version as well (was still using
org.sonatype.aether).

---

https://issues.apache.org/jira/browse/MCOMPILER-472

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -63,7 +63,7 @@ under the License.
   </distributionManagement>
 
   <properties>
-    <mavenVersion>3.0</mavenVersion>
+    <mavenVersion>3.2.5</mavenVersion>
```
[the remark this answers is not in the document]
Comment (slachiewicz) [reply pom.xml:66]: https://issues.apache.org/jira/browse/MCOMPILER-476

**Review by michael-o — APPROVED** [review]
LGTM, but should be split into two commits for those two tickets.

**Comment by alitokmen** [comment]
@rfscholte and @cstamas I have the impression this made it to the Maven Compiler plugin's 3.9.0 release yet https://issues.apache.org/jira/secure/ReleaseNote.jspa?version=12345214&styleName=Html&projectId=12317225&Create=Create&atl_token=A5KQ-2QAV-T4JA-FDED_cb37d11051cd6e0fa49b100de475dff5df5719b7_lout doesn't list https://issues.apache.org/jira/browse/MCOMPILER-472

Is this expected? I found it inconsistent, nevertheless perhaps I missed something.

**Comment by michael-o** [comment]
@alitokmen 

There is no commit on master for this. Why do you think so?

**Comment by alitokmen** [comment]
> @alitokmen
> 
> There is no commit on master for this. Why do you think so?

As Dependabot upgraded our project’s dependency to the Maven Compiler Plugin 3.9.0 our CI popped up the message that the  new version of the Maven Compiler Plugin requires Maven 3.2.5 onwards. The only commit I saw related was this one, hence my wondering.

**Comment by michael-o** [comment]
> > @alitokmen
> > There is no commit on master for this. Why do you think so?
> 
> As Dependabot upgraded our project’s dependency to the Maven Compiler Plugin 3.9.0 our CI popped up the message that the new version of the Maven Compiler Plugin requires Maven 3.2.5 onwards. The only commit I saw related was this one, hence my wondering.

Can you share that issue?

**Comment by slachiewicz** [comment]
07d701ec50553a2d5da85a8466f15f921624c74f #76

**Comment by alitokmen** [comment]
> > > @alitokmen
> > > There is no commit on master for this. Why do you think so?
> > 
> > 
> > As Dependabot upgraded our project’s dependency to the Maven Compiler Plugin 3.9.0 our CI popped up the message that the new version of the Maven Compiler Plugin requires Maven 3.2.5 onwards. The only commit I saw related was this one, hence my wondering.
> 
> Can you share that issue?

You see on https://github.com/apache/maven-compiler-plugin/commit/07d701ec50553a2d5da85a8466f15f921624c74f that the changes made it the the 3.9.0 release of the plugin. Unfortunately since I already updated the CI Maven version I lost the error message .(

@slachiewicz I see you updated https://issues.apache.org/jira/browse/MCOMPILER-472 as _Closed fixed_, could you please also set the _Fix version_ to `3.9.0`?

**Comment by jira-importer** [comment]
Resolve #682

**Pull request closed without merging**